tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,252 @@
+#include <map>
+
+#include "TFile.h"
+#include "TChain.h"
+#include "TH1.h"
+#include "TError.h"
+
+/** \class TSympleAnalysis
+A TSimpleAnalysis object permit to, given an input file,
+create an .root file in which are saved histograms
+that the user wants to create.
+*/
+
+const std::string TSimpleAnalysis::kCutIntr="if";
+
+////////////////////////////////////////////////////////////////////////////////
+/// Delete white spaces in a string
+///
+/// param[in] line line read from the input file
+
+static bool DeleteSpaces(std::string& line)
+{
+   std::size_t firstNotSpace = line.find_first_not_of(" ");
+   if (firstNotSpace != std::string::npos)
+      line = line.substr(firstNotSpace, line.size()-firstNotSpace);
+   std::size_t lastNotSpace = line.find_last_not_of(" ");
+   if (lastNotSpace != std::string::npos)
+      line = line.substr(0, lastNotSpace+1);
+   return 1;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// It handles the expression lines of the input file in order to pass the
+/// elements to the members of the object
+///
+/// param[in] line line read from the input file or the expression passed to the constructor
+
+std::string TSimpleAnalysis::HandleExpressionConfig(std::string& line)
+{
+   std::size_t equal = line.find("=");
+   if (equal == std::string::npos)
+      return "Error: missing '='";
+   std::size_t cutPos = line.find(kCutIntr, equal);
+   std::string histName = line.substr(0, equal);
+   if (histName.empty())
+      return "Error: no histName found";
+   DeleteSpaces(histName);
+
+   std::string histExpression = line.substr(equal+1, cutPos-equal-1);
+   if (histExpression.empty())
+      return "Error: no expression found";
+   DeleteSpaces(histExpression);
+
+   std::string histCut;
+   if (cutPos != std::string::npos)
+      histCut = line.substr(cutPos+kCutIntr.size());
+
+   auto check = fHists.insert(std::make_pair((const std::string&)histName,
+                                             std::make_pair(histExpression, histCut)));
+   if (!check.second)
+      return "Name of the histogram already existing";
+   return "";
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case with the input file
+///
+/// \param[in] kFile name of the input file that has to be read
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& file): fInputName(file)
+{}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case of command line parsing arguments
+///
+/// \param[in] output name of the output file
+/// \param[in] inputFiles name of the input .root files
+/// \param[in] name name of the tree
+/// \param[in] expressions what is showed in the histograms
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& output, const std::vector<std::string>& inputFiles,
+                                 const std::string& name, std::vector<std::string> expressions):
```

Comment:
Rename `name` to `treename`, make it the last argument, and provide a default value of `""`.